### ISSUE TITLE
JavadocMethodCheck: swap scope to access modifiers

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#7417]
Split from https://github.com/checkstyle/checkstyle/issues/3511 ,

Swap scope and exclude scope to access modifier in the check JavadocMethodCheck.

**Migration Notes:**
Property excludeScope was dropped and scope property was from
https://checkstyle.org/property_types.html#scope to accessModifiers of type https://checkstyle.org/property_types.html#access_modifiers .

The change to `accessModifiers` basically combines `scope` and `excludeScope` into 1 field and allows you specify the exact modifiers you want to validate methods with. Scope ambiguously only let you choose 1 specific modifier, and any other modifers above it in our type documentation were included, regardless if you wanted to include them or not. With `accessModifiers` you can now write out any and all modifiers you wish to cover, with no ambiguity.

Examples:

scope => accessModifiers
`private` => `private`, `package`, `protected`, and `public`
`package` => `package`, `protected`, and `public`
`protected` => `protected`, and `public`
`public` => `public`

`excludeScope` only allowed you to remove that 1 specific modifier. So `scope` of `private` and `excludeScope` of `package` included the modifiers `private`, `protected`, and `public`. With access modifiers you can have the same config with:
````
<module name="JavadocMethod">
  <property name="accessModifiers" value="private, protected, public"/>
</module>
````

Scope did support anonymous inner classes. Since that is technically not a modifier, it is no longer supported and there is workaround.

-------

**MIGRATION NOTES**:
old config example:
```xml
        <module name="JavadocMethod">
            <property name="scope" value="public"/>
        </module>
```
should be changed to:
```xml
        <module name="JavadocMethod">
            <property name="accessModifiers" value="public"/>
        </module>
```

old config example:
```xml
        <module name="JavadocMethod">
            <property name="scope" value="protected"/>
        </module>
```
should be changed to:
```xml
        <module name="JavadocMethod">
            <property name="accessModifiers" value="public, protected"/>
        </module>
```

**Comment by shashwat70** [comment]
I'm on it!

**Comment by rnveach** [comment]
@shashwat70 If you want to finish this PR, go ahead as I have no time. Feel free to start with the changes I had in my PR at https://github.com/checkstyle/checkstyle/pull/7418 so you don't have to start from scratch.

I recommend taking my name out of the commit and ensure it only has your name.

**Comment by shashwat70** [comment]
I just read all the conversations in the PR. It looks as if most of the stuff is resolved. I believe https://github.com/checkstyle/checkstyle/pull/7418#discussion_r368346196 is all that is left. Can you confirm? @rnveach

**Comment by nmancus1** [comment]
PR's opened in the following projects to update their Checkstyle config:
equalsverifier: https://github.com/jqno/equalsverifier/pull/416
xwiki-commons: https://github.com/xwiki/xwiki-commons/pull/140
pmd build-tools: https://github.com/pmd/build-tools/pull/18

**Comment by dkfellows** [comment]
It's a shame you didn't fix the other users of `scope` in the area at the same time.

**Comment by XenoAmess** [comment]
What be the meaning for this change?

Taking consider that this change will break already-exist checkstyles files in nearly ALL java repos.

**Comment by romani** [comment]
Yes, it is good for future.
Nobody force you to update checkstyle version, so you should be ok until you need some new feature in new releases.

**Comment by shardt68** [comment]
Hello Nick,
would you please deprecate changes like this in first place and delete the old property in a major version later. 
We have old released projects containing the config. And newer ones where we wold like to switch. Now we have to work with 2 different eclipses with different plugins. 
Regards, Sven

**Comment by romani** [comment]
@DanySK, please try to use same version of plugins in your courses. Maintainers spend a lot of time to have plugin version for each version of library.
We are changing not only configuration but behavior too, so different versions of library might give different violations.
Our release notes are always correct and shows breaking changes https://checkstyle.org/releasenotes.html

As we are library, we do not know where are used so it is challenging to print deprecation warning.

I still highly recommend to use strict config and version, and upgrade only if you have time to deal with changes. We release frequently so you can always upgrade up to version that has breaking changes.

**Comment by DanySK** [comment]
@romani you are right, but, unfortunately, I've got limited control over what bachelor students install on their own PCs, and very little over what technicians prepare for us in the labs as well. They do pull stuff from the Eclipse Marketplace, so it is up to me to try to cope with multiple versions (ideally, by providing a conf file that works throughout the latest year or so releases).

The change has never been a big deal for my FOSS projects, I know how to lock versions and make sure that upgrades work as expected (also, I maintain [my own Java QA plugin](https://github.com/DanySK/gradle-java-qa) that takes care of selecting the correct version of Checkstyle for multiple projects of mine).
I also agree that the changelogs and the docs are great, so please, do not intend my previous comment as an accusation of not doing enough.

What I am suggesting is to consider whether it is possible to smoothen the update procedure, so that there is enough time to adapt. Maybe it is not possible to have clearer deprecations, I'm not sure. In the extreme case, exposing an explicit API for "rules that will change in future releases" could help the developers expose warnings downstream.
I understand that this would be a further development/maintenance burden, though.
